src/rgw/rgw_multi.h

```diff
@@ -60,12 +60,18 @@ class RGWMultiXMLParser : public RGWXMLParser
 class MultipartMetaFilter : public RGWAccessListFilter {
 public:
   MultipartMetaFilter() {}
-  bool filter(string& name, string& key) override {
-    int len = name.size();
-    if (len < 6)
+  bool filter(const string& name, string& key) override {
+    size_t len = name.size();
+
+    // don't count null terminator
+    constexpr size_t MP_META_SUFFIX_LEN = sizeof(MP_META_SUFFIX) - 1;
```

Comment:
this is odd and C-like; could we use std::string statics and take length()?

Reply:
Yeah, I was trying to push constexpr as far as I could and strings are not compatible. But keeping the suffix as a const string and then using a const static size_t for the length works and only has to be computed once for the entire run of the rgw.